nsiqcppstyle_exe.py

```diff
@@ -78,91 +77,138 @@ def Usage():
                 to create one (e.g., in a read-only file system)
   --var=key: value,key: value
                 provide the variables to customize the rule behavior.
-  --list-rules / -r  Show all rules available.
-                Add file extensions to be counted as assigned languages.
-  -s            Assign Filter scope name to be applied in this analysis
-  --output=     output format 'emacs', 'vs7', 'csv', 'xml' and 'eclipse'. Default value is vs7
-                emacs, vs7, eclipse output the result on the stdout in the form
-                that each tool recognizes.
-                csv and xml outputs the result on the file "nsiqcppstyle_result.csv"
-                "nsiqcppstyle_result.xml" respectively, if you don't provide -o option.
-  --ci          Continuous Integration mode. If this mode is on, this tool only reports summary.
-  --quiet / -q  Quiet mode. If this mode is on, this tool only reports errors.
+
 
 * nsiqcppstyle reports coding standard violations on C/C++ source code.
-* In default, it doesn't apply any rules on the source. If you want to apply rule,
-  they should be provided in the filefilter.txt file in following form.
+
+* By default, it doesn't apply any rules on the source. If you want to apply rule, they should be provided in the 'filefilter.txt' file in following form.
   ~ RULENAME
 
-* You can customize the rule behavior by providing --var=key: value pair when executing
-  tool and you can put it in the filefilter.txt. The format is following.
+* You can customize the rule behavior by providing --var=key: value pair when executing tool and you can put it in the 'filefilter.txt'. The format is following.
```

Comment:
As mentioned above, the fact that a single whitespace is mandatory between "key:" and the value was a source of woe for me :smiley:.  I recommend writing the following:

<snippet>
* You can customize the rule behavior by providing the command line option --var="key: value,key: value" string.  Note that the space between "key:" and the value is mandatory. It can be inserted in the 'filefilter.txt' file using the following format:

  % key: value
</snippet>

Reply:
I'm glad to know this is a real pain point. I intend to add more documentation and move this to either:
* 1 --var for each key aka `--var key1:value --var key2:value`
* Just a single --var aka `--var key1:value key2:value`

Which one is preferred?

Reply:
I'm partial towards 1st because it works better with the N files which can be provided as argument

Reply:
If you're asking, I think the second option (i.e., just a single pair per --var) is the simpler and preferable one.  Make sure to update the help text accordingly. :smiley:

Reply:
I hadn't seen your previous comment two minutes before mine :smiley:.  In deference to the maintainer, go for it.  Just make sure the help text makes it cleat regarding the mandatory space, where the comma goes, and that the entire --var string should be quoted as per Windows or Linux.